pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -192,7 +195,10 @@ def _run_binsearch_scaling(
             else:
                 new_size, changed = _adjust_batch_size(trainer, batch_arg_name, factor=2.0, desc='succeeded')
 
-            if not changed:
+            if changed:
+                # Force the train dataloader to reset as the batch size has changed
+                trainer.train_dataloader = None
```

Comment:
what's the reason for not calling the reset method?

Reply:
I guess that would work too. Let's change it.

Reply:
Dear @awaelchli , thanks for your suggestion. Updated!